sunpy/coordinates/frames.py

```diff
@@ -587,6 +588,13 @@ def make_3d(self):
 
             d = np.fmin(d, dd) if self._spherical_screen['only_off_disk'] else dd
 
+        stack_trace = traceback.format_stack()
+        # _draw_grid is the specific function called in WCSAxes
+        if not any(["_draw_grid" in string for string in stack_trace]) and np.all(np.isnan(d)) and np.any(np.isfinite(cos_alpha)):
```

Comment:
I haven't used `stack_trace` before but can we not put a fully qualified path here or something else mentioning `wcsaxes` in some way?

Reply:
You mean check for the file name and etc?

Reply:
yeah, or class at least?

Reply:
I can add part of the file path.

Reply:
Done it, but it's probably worse now.

Reply:
Nope, broke it.